core/src/main/scala/kafka/tools/StorageTool.scala

```diff
@@ -53,12 +61,23 @@ object StorageTool extends Logging {
             throw new TerseFailure(s"Must specify a valid KRaft metadata version of at least 3.0.")
           }
           val metaProperties = buildMetadataProperties(clusterId, config.get)
+          val metadataRecords : ArrayBuffer[ApiMessageAndVersion] = ArrayBuffer()
+          getUserScramCredentialRecords(namespace).foreach { 
+            if (!metadataVersion.isScramSupported()) {
+              throw new TerseFailure(s"SCRAM is only supported in metadataVersion IBP_3_5_IV0 or later.");
```

Comment:
Please add a test case that checks that this failure case works

Reply:
Done.